src/content_mapblock.cpp

```diff
@@ -841,19 +841,19 @@ void MapblockMeshGenerator::drawPlantlikeQuad(float rotation, float quad_offset,
 	bool offset_top_only)
 {
 	v3f vertices[4] = {
-		v3f(-scale, -BS / 2, 0),
-		v3f( scale, -BS / 2, 0),
-		v3f( scale, -BS / 2 + scale * 2, 0),
 		v3f(-scale, -BS / 2 + scale * 2, 0),
+		v3f( scale, -BS / 2 + scale * 2, 0),
+		v3f( scale, -BS / 2, 0),
+		v3f(-scale, -BS / 2, 0),
 	};
 	if (random_offset_Y) {
 		PseudoRandom yrng(face_num++ | p.X << 16 | p.Z << 8 | p.Y << 24);
 		offset.Y = BS * ((yrng.next() % 16 / 16.0) * 0.125);
 	}
-	int offset_first_index = offset_top_only ? 2 : 0;
+	int offset_first_count = offset_top_only ? 2 : 4;
```

Comment:
My comment was only asking for a change of variable names, for the code as it currently is the variable is the number of offset verexes, whereas before it was the first idex offset.
I agree with SmallJoker, one loop, 3rd example.